system_modules/napvideo/src/video.cpp

```diff
@@ -737,6 +742,12 @@ namespace nap
 				mFrameDataAvailableCondition.notify_all();
 				VIDEO_DEBUG_LOG("push frame (stream %d): pkt_pos: %d, dts: %d, pts: %d", mStream, new_frame.mFrame->pkt_pos, new_frame.mFrame->pkt_dts, new_frame.mFrame->pkt_pts);
 			}
+
+            // sleep the thread for a bit
+            // this prevents a very rare race condition that can happen while seeking
+            // a seekstart packet can be pushed onto the queue but the decode thread is so fast that it processes the packet before waitSeekStartPacketProcessed is called
+            // this can happen with black frames in the videos and fast CPU's
+            std::this_thread::sleep_for(std::chrono::milliseconds(1));
```

Comment:
Yeah, this is probably not what we want. Can you find a proper way to wait for the packet to be decoded if `waitSeekStartPacketProcessed` is  required to be called? Maybe a `std::condition_variable`?

Reply:
I addressed this in commit -> https://github.com/napframework/nap/pull/28/commits/3832364b65a36be0f8979370969c8785b552425d

Basically, emptying the seek frame queue before waiting for the mSeekStartProcessedEvent prevents the potential race condition. There is a possibility the mSeekStartProcessedEvent doesn't get called when the seek frame queue is full from a previous (interrupted) seek, but the frames don't get popped because we're waiting for the mSeekStartProcessedEvent to fire

Reply:
`There is a possibility the mSeekStartProcessedEvent doesn't get called when the seek frame queue is full from a previous (interrupted) seek, but the frames don't get popped because we're waiting for the mSeekStartProcessedEvent to fire`

What is waiting for the event to fire exactly?

Reply:
One thread is has a wait for the mSeekStartProcessedEvent but the thread that sets the event is waiting for the seek frame queue to get < 16 before proceeding to decodeframes, also allowing the seekStartPacket to be handled.

Emptying the seek frame queue there prevents this from happening.